sys/linux/auto.txt

```diff
@@ -1059,63 +1039,271 @@ syz_genetlink_get_family_id$auto_tcp_metrics(name ptr[in, string["tcp_metrics"]]
+times$auto(tbuf ptr[inout, tms$auto_record]) (automatic)
+tkill$auto(pid int32, sig int32) (automatic)
+truncate$auto(path ptr[inout, int8], length int32) (automatic)
+ugetrlimit$auto(rsrc int32, rlim ptr[inout, rlimit$auto_record]) (automatic)
+umask$auto(mask int32) (automatic)
+umount$auto(name ptr[inout, int8]) (automatic)
+umount2$auto(name ptr[inout, int8], flags int32) (automatic)
+uname$auto(name ptr[inout, new_utsname$auto_record]) (automatic)
+unlink$auto(pathname ptr[inout, int8]) (automatic)
+unlinkat$auto(dfd int32, pathname ptr[inout, int8], flag int32) (automatic)
+unshare$auto(unshare_flags int32) (automatic)
+userfaultfd$auto(flags int32) (automatic)
+ustat$auto(dev int32, ubuf ptr[inout, ustat$auto_record]) (automatic)
+utime$auto(filename ptr[inout, int8], t ptr[inout, old_utimbuf32$auto_record]) (automatic)
+utimensat$auto(dfd int32, filename ptr[inout, int8], t ptr[inout, old_timespec32$auto_record], flags int32) (automatic)
+utimensat_time64$auto(dfd int32, filename ptr[inout, int8], utimes ptr[inout, __kernel_timespec$auto_record], flags int32) (automatic)
+utimes$auto(filename ptr[inout, int8], t ptr[inout, old_timeval32$auto_record]) (automatic)
+vmsplice$auto(fd int32, uiov ptr[inout, iovec$auto_record], nr_segs int32, flags int32) (automatic)
+wait4$auto(upid int32, stat_addr ptr[inout, int32], options int32, ru ptr[inout, rusage$auto_record]) (automatic)
+waitid$auto(which int32, upid int32, infop ptr[inout, siginfo$auto_record], options int32, ru ptr[inout, rusage$auto_record]) (automatic)
+waitpid$auto(pid int32, stat_addr ptr[inout, int32], options int32) (automatic)
+write$auto(fd int32, buf ptr[inout, int8], count int32) (automatic)
+writev$auto(fd int32, vec ptr[inout, iovec$auto_record], vlen int32) (automatic)
 _ = __NR_mmap2
 
+XFRMA_COADDR$auto_record [
+	a4	int32
+	a6	array[int32, 4]
+	in6	in6_addr$auto_record
+]
+
+XFRMA_SRCADDR$auto_record [
+	a4	int32
+	a6	array[int32, 4]
+	in6	in6_addr$auto_record
+]
+
+__aio_sigset$auto_record {
+	sigmask		ptr[inout, sigmask$auto_record]
+	sigsetsize	int32
+}
+
+__kernel_itimerspec$auto_record {
+	it_interval	__kernel_timespec$auto_record
+	it_value	__kernel_timespec$auto_record
+}
+
+__kernel_old_itimerval$auto_record {
+	it_interval	__kernel_old_timeval$auto_record
+	it_value	__kernel_old_timeval$auto_record
+}
+
+__kernel_old_timeval$auto_record {
+	tv_sec	int32
+	tv_usec	int32
+}
+
+__kernel_sockaddr_storage$auto_record {
+	__kernel_sockaddr_storage_0	__kernel_sockaddr_storage_0$auto_record
+}
+
+__kernel_sockaddr_storage_0$auto_record [
+	__kernel_sockaddr_storage_0_0	__kernel_sockaddr_storage_0_0$auto_record
+	__align				ptr[inout, void]
+]
+
+__kernel_sockaddr_storage_0_0$auto_record {
+	ss_family	int16
+	__data		array[int8, 126]
+}
+
+__kernel_timespec$auto_record {
+	tv_sec	intptr
+	tv_nsec	intptr
+}
+
+__kernel_timex$auto_record {
+	modes			int32
+	__kernel_timex_1	int32
```

Comment:
Nit: pads are better be `const[int32, 0]` so that the fuzzer does not mutate their values

Reply:
Are all anonymous bitfields pads? Is there anything else that's considered padding?

Reply:
Yes, anonymous bitfields are pads. As I mentioned in some other discussion, I'd also consider to be a pad everything that was explicitly called a pad.